### PR TITLE
RHIDP-3202: Populate Upgrading RHDH title with new/existing content

### PR DESCRIPTION
**Pull request by linfraze**
This PR includes the following doc changes:

- Removes upgrading information from the RHDH 1.2 Release Notes 
- Moves existing content about upgrading with the Helm chart to the _Upgrading RHDH_ title
- Creates new content about upgrading with the Operator in the _Upgrading RHDH_ title

**Version(s):**
1.2.2

**Issue:**
https://issues.redhat.com/browse/RHIDP-3202

**Link to docs preview:**
https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-406/upgrade-rhdh/

**Reviews:**
- [x] SME: @rm3l @Fortune-Ndlovu 
- [x] QE: @zdrapela 
- [x] Docs review: @gwynnemonahan 

Will merge 2 commits to 1.2.x and cherrypick 1 to main.

### CODE REVIEW

**Review by Fortune-Ndlovu — APPROVED** [review]
/lgtm

**Review by zdrapela — APPROVED** [review]
LGTM

File: modules/upgrade/proc-upgrade-rhdh-helm.adoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
++
+[source]
+----
+oc login -u <user> -p <password> https://api.<HOSTNAME>:6443
+oc project <your_namespace>
+----
+
+. For a new version of the {product-short} Helm chart, run the following upgrade command:
++
+[source,subs="attributes+"]
```
Comment (gwynnemonahan) [inline modules/upgrade/proc-upgrade-rhdh-helm.adoc:40]: Should this be [source, yaml, subs="attributes+"]

File: modules/upgrade/proc-upgrade-rhdh-helm.adoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+It might take a few minutes to delete the resources in the older versions and to start the newer versions of the {product-short} pods.
+====
+
+. Close all open {product-short} web pages, and log in again to verify that the upgrade was successful.
+
+* {ocp-short} CLI
+
+. Log in to the {ocp-short} cluster as the cluster administrator and switch to the project or namespace in which {product-short} was installed.
++
+[source]
```
Comment (gwynnemonahan) [inline modules/upgrade/proc-upgrade-rhdh-helm.adoc:32]: source....should this be [source, terminal]

File: modules/upgrade/proc-upgrade-rhdh-helm.adoc [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies
+//
+
+:_mod-docs-content-type: PROCEDURE
+[id="proc-upgrade-rhdh-helm_{context}"]
+
+= Upgrading the {product} Helm Chart
+
+You can upgrade to a later version of {product} in {ocp-short} by using either the web console or the CLI.
```
Comment (gwynnemonahan) [inline modules/upgrade/proc-upgrade-rhdh-helm.adoc:9]: Perhaps more a point of clarification: is the Helm chart the same as the RHDH? So if you update the Helm chart by either the web console or the CLI, you are updating Red Hat Developer Hub, correct? 

The title suggests the user is only upgrading the Helm chart. The first sentence suggests the users is upgrading Red Hat Developer Hub, so wanted to check if there is a difference.

**Review by gwynnemonahan — COMMENTED** [review]
Couple of style things. Otherwise, LGTM.

File: modules/upgrade/proc-upgrade-rhdh-helm.adoc (hunk 1 above)
Comment (linfraze) [reply modules/upgrade/proc-upgrade-rhdh-helm.adoc:40]: The text in this file was not changed nor created under this PR, as the scope of this PR is only a relocation of this existing content. IMO, this code block is not even necessary; we might consider removing it in the future.

File: modules/upgrade/proc-upgrade-rhdh-helm.adoc (hunk 2 above)
Comment (linfraze) [reply modules/upgrade/proc-upgrade-rhdh-helm.adoc:32]: The text in this file was not changed nor created under this PR, as the scope of this PR is only a relocation of this existing content. IMO, this code block is not even necessary; we might consider removing it in the future.

File: modules/upgrade/proc-upgrade-rhdh-helm.adoc (hunk 1 above)
Comment (linfraze) [reply modules/upgrade/proc-upgrade-rhdh-helm.adoc:40]: Yes, it should be

**Review by gwynnemonahan — APPROVED** [review]

File: modules/upgrade/proc-upgrade-rhdh-helm.adoc (hunk 3 above)
Comment (linfraze) [reply modules/upgrade/proc-upgrade-rhdh-helm.adoc:9]: I agree this is confusing. When I asked the SMEs this same question (specifically for the Operator), the answer I received was along the lines of _yes, you update the Operator to update RHDH_.

**Pull request merged**